design/index_files.md

```diff
@@ -0,0 +1,64 @@
+# On-Disk persistent index files
+
+Authored: Jan 22, 2024
+Author: Thor Hansen (@thorfour)
+
+## Abstract
+
+FrostDB currently uses temporary index files to store Parquet data. These are lost on shutdown, and need to be recreated from snapshots and WAL.
+Furthermore snapshots are less effecient than necessary because snapshotting copies Parquet data that is already on disk, and writes it to a different file (snapshot file)
+when we could just reference the index file instead.
+
+## Proposal
+
+Instead FrostDB should use persistent index files that don't requrie snapshots to copy data but instead hard link the files in the snapshot directory for recovery.
+
+Index files have the format of:
+
+|--------------------|
+|   <ParquetFile>    |
+|   <uint64 size>    |
+|   <ParquetFile>    |
+|   <uint64 size>    |
+|   <ParquetFile>    |
+|   <uint64 size>    |
+|--------------------|
+
+Where Parquet files are appending togehter with a uint64 that represents the number of bytes in the previous ParquetFile.
+Each Parquet file represents a single Part in the LSM index. The Part's transaction id is stored in the Parquet file with a metadata key of 'compaction_tx'
```

Comment:
"Each parquet file represents a single part in the LSM index" this is no longer true, right?

Reply:
It's still true. A index file has many Parquet files/Parts in it, but each Parquet file is a part.